src/shapes/active_selection.class.ts

```diff
@@ -1,191 +1,174 @@
-          parent._exitGroup(object);
-          object.__owningGroup = parent;
-        }
-        this._enterGroup(object, removeParentTransform);
-        return true;
-      },
+  /**
+   * @private
+   * @param {FabricObject} object
+   * @param {boolean} [removeParentTransform] true if object is in canvas coordinate plane
+   * @returns {boolean} true if object entered group
+   */
+  enterGroup(object: FabricObject, removeParentTransform?: boolean) {
+    if (object.group) {
+      //  save ref to group for later in order to return to it
+      const parent = object.group;
+      parent._exitGroup(object);
+      object.__owningGroup = parent;
+    }
+    this._enterGroup(object, removeParentTransform);
+    return true;
+  }
 
-      /**
-       * we want objects to retain their canvas ref when exiting instance
-       * @private
-       * @param {fabric.Object} object
-       * @param {boolean} [removeParentTransform] true if object should exit group without applying group's transform to it
-       */
-      exitGroup: function (object, removeParentTransform) {
-        this._exitGroup(object, removeParentTransform);
-        var parent = object.__owningGroup;
-        if (parent) {
-          //  return to owning group
-          parent.enterGroup(object);
-          delete object.__owningGroup;
-        }
-      },
+  /**
+   * we want objects to retain their canvas ref when exiting instance
+   * @private
+   * @param {FabricObject} object
+   * @param {boolean} [removeParentTransform] true if object should exit group without applying group's transform to it
+   */
+  exitGroup(object: FabricObject, removeParentTransform?: boolean) {
+    this._exitGroup(object, removeParentTransform);
+    const parent = object.__owningGroup;
+    if (parent) {
+      //  return to owning group
+      parent.enterGroup(object);
+      delete object.__owningGroup;
+    }
+  }
 
-      /**
-       * @private
-       * @param {'added'|'removed'} type
-       * @param {fabric.Object[]} targets
-       */
-      _onAfterObjectsChange: function (type, targets) {
-        var groups = [];
-        targets.forEach(function (object) {
-          object.group &&
-            !groups.includes(object.group) &&
-            groups.push(object.group);
-        });
-        if (type === 'removed') {
-          //  invalidate groups' layout and mark as dirty
-          groups.forEach(function (group) {
-            group._onAfterObjectsChange('added', targets);
-          });
-        } else {
-          //  mark groups as dirty
-          groups.forEach(function (group) {
-            group._set('dirty', true);
-          });
-        }
-      },
+  /**
+   * @private
+   * @param {'added'|'removed'} type
+   * @param {FabricObject[]} targets
+   */
+  _onAfterObjectsChange(type: 'added' | 'removed', targets: FabricObject[]) {
+    // @TODO figure out this change. This part wasn't here before migration.
+    super._onAfterObjectsChange(type, targets);
```

Comment:
I m merging now, but i wonder why we added this line here.
This adds an extra layout call that we weren't doing before?

Reply:
Yes, correct.
It was a bug that I fixed in a pending PR so I decided to put it here